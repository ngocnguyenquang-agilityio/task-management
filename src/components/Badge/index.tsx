import type { ReactNode } from 'react';
import clsx from 'clsx';

type BadgeProps = {
  label: string;
  theme?:
    | 'primary'
    | 'secondary'
    | 'tertiary'
    | 'warning'
    | 'error'
    | 'success';
  icon?: ReactNode;
  customClass?: string;
};

export const Badge = ({
  label,
  theme = 'primary',
  icon,
  customClass,
}: BadgeProps) => {
  const baseClass = 'w-fit flex items-center gap-2 px-2 py-1';

  return (
    <div
      className={clsx(
        baseClass,
        {
          'bg-blue-950': theme === 'primary',
          'bg-indigo-500': theme === 'secondary',
          'bg-purple-500': theme === 'tertiary',
          'bg-amber-400': theme === 'warning',
          'bg-red-500': theme === 'error',
          'bg-emerald-500': theme === 'success',
        },
        customClass,
      )}
    >
      {icon}
      {label}
    </div>
  );
};
