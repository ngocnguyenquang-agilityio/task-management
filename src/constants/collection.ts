export const COLLECTION = {
  USERS: 'users',
  TASKS: 'tasks',
  PROJECTS: 'projects',
  PARTICIPATIONS: 'participations',
};

export const LIMIT_ITEMS = {
  DEFAULT: 10,
  BOARD_PAGE: 6,
  SINGLE_RECORD: 1,
};
