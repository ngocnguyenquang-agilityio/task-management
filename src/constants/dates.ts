export enum DATE_FORMAT {
  Primary = 'MM/DD/YYYY', // Ex: 08/16/2022
  Secondary = 'MMMM DD, YYYY', // Ex: August 01, 2022
  Tertiary = 'YYYY-MM-DD', // Ex: 2024-06-27
  Hour = 'ha', // Ex: 1pm
}
