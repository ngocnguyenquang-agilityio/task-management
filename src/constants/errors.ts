export const ERROR_CODE = {
  EMAIL_EXIST: 'auth/email-already-in-use',
};
