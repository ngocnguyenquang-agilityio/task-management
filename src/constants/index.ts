export * from './dates';
export * from './mocks';
export * from './messages';
export * from './navigations';
export * from './validators';
export * from './schemas';
export * from './collection';
export * from './errors';
export * from './tasks';
export * from './queryParams';
