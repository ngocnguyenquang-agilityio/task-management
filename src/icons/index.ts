export * from './BoardIcon';
export * from './ClockIcon';
export * from './DotIcon';
export * from './HomeIcon';
