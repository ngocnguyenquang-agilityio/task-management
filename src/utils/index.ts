export * from './slugs';
export * from './dates';
export * from './numbers';
export * from './styles';
export * from './validators';
export * from './queryParams';
