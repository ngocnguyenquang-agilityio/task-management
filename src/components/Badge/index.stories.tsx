import type { Meta, StoryObj } from '@storybook/react';

// Components
import { Badge } from './index';
import { ClockIcon } from '@/icons';

const meta = {
  title: 'Components/Badge',
  component: Badge,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    theme: {
      control: 'inline-radio',
      options: [
        'primary',
        'secondary',
        'tertiary',
        'success',
        'error',
        'warning',
      ],
    },
  },
} satisfies Meta<typeof Badge>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    icon: <ClockIcon />,
    label: 'Status',
    theme: 'primary',
    customClass: 'rounded-md text-xs fill-white text-white ',
  },
};
