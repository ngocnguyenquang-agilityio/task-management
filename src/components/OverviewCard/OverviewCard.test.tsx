import { render, screen } from '@testing-library/react';

// Components
import { OverviewCard } from './index';
import { Badge } from '../Badge';

// Constants
import { DATE_FORMAT, MOCK_OVERVIEW_CARD_DATA } from '@/constants';

// Icons
import { ClockIcon } from '@/icons';

// Utils
import { formatDate } from '@/utils';

describe('OverviewCard component', () => {
  test('renders the OverviewCard match snapshot', () => {
    const { container } = render(
      <OverviewCard
        href="/"
        title={MOCK_OVERVIEW_CARD_DATA.title}
        description={MOCK_OVERVIEW_CARD_DATA.description}
        time={formatDate(MOCK_OVERVIEW_CARD_DATA.createdAt, DATE_FORMAT.Hour)}
        isRowDisplay={true}
        badge={
          <Badge
            label="Design Project"
            icon={<ClockIcon />}
            customClass="rounded-md text-white text-sm"
          />
        }
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders the OverviewCard with children', () => {
    render(
      <OverviewCard
        href="/"
        title={MOCK_OVERVIEW_CARD_DATA.title}
        description={MOCK_OVERVIEW_CARD_DATA.description}
        time={formatDate(MOCK_OVERVIEW_CARD_DATA.createdAt, DATE_FORMAT.Hour)}
        isRowDisplay={true}
        badge={
          <Badge
            label="Design Project"
            icon={<ClockIcon />}
            customClass="rounded-md text-white text-sm"
          />
        }
      />,
    );
    expect(screen.getByText('Design Project')).toBeInTheDocument();
    expect(
      screen.getByText(
        formatDate(MOCK_OVERVIEW_CARD_DATA.createdAt, DATE_FORMAT.Hour),
      ),
    ).toBeInTheDocument();
  });
});
