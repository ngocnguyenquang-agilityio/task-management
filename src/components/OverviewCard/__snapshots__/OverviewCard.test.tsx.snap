// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewCard component renders the OverviewCard match snapshot 1`] = `
<div>
  <a
    class="flex flex-col border rounded-lg p-5 dark:bg-zinc-800 dark:border-gray-700 md:flex-row md:gap-4"
    href="/"
  >
    <div
      class="relative block bg-zinc-200 rounded-lg flex-1 md:basis-2/5"
    />
    <div
      class="flex flex-1 flex-col gap-1 dark:text-white min-w-0 md:basis-3/5"
    >
      <div
        class="w-fit flex items-center gap-2 px-2 py-1 bg-blue-950 rounded-md text-white text-sm"
      >
        <svg
          class="w-2.5 h-2.5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 12 12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.59998 4V6L6.84998 7.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
          <path
            d="M3.09998 1.66901C3.85974 1.22946 4.72223 0.998653 5.59998 1.00001C8.36148 1.00001 10.6 3.23851 10.6 6.00001C10.6 8.76151 8.36148 11 5.59998 11C2.83848 11 0.599976 8.76151 0.599976 6.00001C0.599976 5.08951 0.843476 4.23501 1.26898 3.50001"
            stroke-linecap="round"
          />
        </svg>
        Design Project
      </div>
      <div
        class="flex justify-between items-center "
      >
        <div
          class="text-md font-bold truncate"
        >
          Website Design Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pulvinar, nisl sed commodo viverra, urna enim interdum ligula, vitae tempor lorem nibh tincidunt nisi. 
        </div>
        <div
          class="font-normal text-xs text-zinc-500"
        >
          3pm
        </div>
      </div>
      <div
        class="text-xs text-zinc-500 truncate"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut pulvinar, nisl sed commodo viverra, urna enim interdum ligula, vitae tempor lorem nibh tincidunt nisi. Aliquam pellentesque metus non sollicitudin mattis. Cras quis auctor massa, at facilisis massa. Vivamus efficitur eros eget ante aliquet semper. Nunc tempus risus turpis, et sagittis felis consectetur eu. Integer luctus felis sed neque rutrum mattis. Quisque molestie tincidunt ligula, id fermentum dui dignissim quis. Nulla facilisi. Integer eget nulla a velit aliquet commodo vitae et leo. Aliquam erat volutpat. Cras nec arcu at dui blandit maximus. Curabitur rhoncus convallis arcu, et condimentum diam facilisis sollicitudin. Quisque in gravida nunc, sit amet semper tortor. Fusce dignissim pretium odio sit amet sagittis.
      </div>
    </div>
  </a>
</div>
`;
